core/java/com/google/instrumentation/stats/MeasurementDescriptor.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * MeasurementDescriptor.
+ *
+ * <p>Note: MeasurementDescriptor names are {@link String}s with enforced restrictions.
+ */
+public final class MeasurementDescriptor {
+  public static final int MAX_LENGTH = StringUtil.MAX_LENGTH;
+
+  /**
+   * Name of measurement, e.g. rpc_latency, cpu. Must be unique.
+   */
+  public final String name;
+
+  /**
+   * Detailed description of the measurement, used in documentation.
+   */
+  public final String description;
+
+  /**
+   * The units in which {@link MeasurementDescriptor} values are measured.
+   */
+  public final MeasurementUnit unit;
+
+  /**
+   * Construtor.
+   */
+  public MeasurementDescriptor(String name, String description, MeasurementUnit unit) {
+    this.name = StringUtil.sanitize(name);
+    this.description = description;
+    this.unit = unit;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return (obj instanceof MeasurementDescriptor)
+        && name.equals(((MeasurementDescriptor) obj).name);
+  }
+
+  @Override
+  public int hashCode() {
+    return name.hashCode();
+  }
+
+  @Override
+  public String toString() {
+    return name;
+  }
+
+  /**
+   * Fundamental units of measurement.
+   */
+  public enum BasicUnit {
+    UNKNOWN,
```

Comment:
Remove UNKNOWN - this is only in the proto format to distinguish it as uninitialized. In this context (people using this API) it should never be set/used.